From 2dc43dc95dd9e6e79ad0b8acaa10a60155f0a974 Mon Sep 17 00:00:00 2001
Message-ID: <2dc43dc95dd9e6e79ad0b8acaa10a60155f0a974.1688558232.git.stefan@agner.ch>
In-Reply-To: <2950cb121e2e2e2dbfb4c02b46f9344427101cb9.1688558232.git.stefan@agner.ch>
References: <2950cb121e2e2e2dbfb4c02b46f9344427101cb9.1688558232.git.stefan@agner.ch>
From: David Rheinsberg <david@readahead.eu>
Date: Wed, 5 Jul 2023 09:52:49 +0200
Subject: [PATCH] launch/config: assert on expat initialization

Ensure the expat initialization did not indicate failure. With the
setup we use, there cannot be any failure, but ensure this is true in
case other setups end up being used in the future.

Signed-off-by: David Rheinsberg <david@readahead.eu>
---
 src/launch/config.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/launch/config.c b/src/launch/config.c
index 3dc3f8f..62682aa 100644
--- a/src/launch/config.c
+++ b/src/launch/config.c
@@ -1230,6 +1230,7 @@ void config_parser_init(ConfigParser *parser) {
         *parser = (ConfigParser)CONFIG_PARSER_NULL(*parser);
 
         parser->xml = XML_ParserCreate(NULL);
+        c_assert(parser->xml);
 }
 
 /**
-- 
2.41.0

